openmdao/docs/theory_manual/total_derivs/vectorized.rst

```diff
@@ -0,0 +1,93 @@
+OpenMDAO needs two vectors for each linear solve (one for the solution and one for the right hand side), using 0.16 MB for the pair.
+
+A non-vectorized solve would require just the single pair of vectors,
+but a vectorized derivative solve needs 1000 right hand side vectors and 1000 solution vectors all at once, using 160 MB of memory.
+
+On modern computers 160 MB of memory is not a significant amount, especially if it comes with the benefit of a large computational savings.
+However, the above example only involved 2 components and hence the cost of the recursive solves won't be significant either.
+
+So instead, consider what would happen if you had the same 5 design variables and 5 constraints of size 1000, but now the model is comprised of 15 different components connected in a feed-forward manner arranged in some complex hierarchy.
+Each component will have its own 5 output variables, giving a total of 75 variables of size 1000 each.
+In this larger model each vector is now of size 75000 using 0.6 MB of memory.
+So a non-vectorized solve will use 1.2 MB of memory but the vectorized solve will now use 1200 MB (1.2 GB) of memory.
+
+This illustrates how the memory cost of vectorization grows with two key factors:
+
+    #. The number of output variables in a model
+    #. The size of the output variable you want to vectorize the total derivative solve for
+
+Ultimately you have to weigh the memory cost and compute savings to determine if this feature is good for your specific use case.
+
+----------------------------------------------------------------
+Usage With Components That Use Assembled Jacobians
+----------------------------------------------------------------
+
+If all components in your model are using the :ref:`compute_partials <comp-type-2-explicitcomp>` or :ref:`linearize <comp-type-3-implicitcomp>` methods to provide OpenMDAO with their partial derivatives, then you do not need to do anything else in order to support vectorized derivatives solves.
+
+Internally, OpenMDAO will switch from doing matrix-vector products to vectorized matrix-matrix products.
+Essentially in either case, if you stored the partial derivative Jacobian in a variable :code:`A`, and your vector as variable :code:`b` will always do
+
+.. code::
+
+    c = A.dot(b)
+
+This NumPy syntax works whether b is a vector or a collection of vectors that have been stacked together (vectorized), which is why you don't need to do anything differently when using assembled Jacobians.
+
+----------------------------------------------------------------
+Usage With Components That Use Matrix-Free Partial Derivatives
+----------------------------------------------------------------
+
+If you have any components that use the matrix-free APIs,
+
+    #. :ref:`compute_jacvec_product<comp-type-2-explicitcomp>`
+    #. :ref:`apply_linear<comp-type-3-implicitcomp>`
+
+Then you need to implement an additional methods in order to use vectorized derivative solves.
```

Comment:
"an additional method" -> an additional method 
(or possibly "one or more additional methods" if we support both together. I think in theory we do, but I don't think I've tried it.)